lib/TPP/TransformUtils.cpp

```diff
@@ -208,13 +208,83 @@ bool isBlockedConvolution(Operation *op) {
   return tpp::utils::hasMatmulBody(linalgOp);
 }
 
-// TODO: Check indexing maps and iterator types. They should
-// match the one of a packed matmul.
+static llvm::SmallDenseSet<unsigned> getPreservedDims(AffineMap map) {
+  assert(map.isProjectedPermutation() &&
+         "expected map to have projected permutations");
+  llvm::SmallDenseSet<unsigned> preservedDims;
+  for (auto expr : map.getResults())
+    preservedDims.insert(expr.cast<AffineDimExpr>().getPosition());
+  return preservedDims;
+}
+
 bool isBlockedMatmul(Operation *op) {
-  if (!isa<linalg::LinalgOp>(op))
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(op);
+  if (!linalgOp)
     return false;
-  linalg::LinalgOp linalgOp = cast<linalg::LinalgOp>(op);
-  return tpp::utils::hasMatmulBody(linalgOp);
+  if (linalgOp.getNumDpsInputs() != 2 || linalgOp.getNumDpsInits() != 1)
```

Comment:
We need to stop using magic constants, but this is for another PR.

Reply:
I want to make sure that I am dealing with a matul so two inputs 1 output.

Reply:
I get what you want, but we keep spreading those checks against magic constants all over the place and that's not good practice. It leads to either misunderstanding what they're checking or copying and pasting elsewhere and failing to update all copies of the same check.

My point is that we need to common those up: Checks, magic constants, semantics of what we expect of operations, etc. As we ramp up the matching for more TPP ops, this will get worse very quickly.